api/src/main/java/javax/application/configuration/Configuration.java

```diff
@@ -65,7 +63,7 @@
      *
      * @return the value associated with {@code key}, or {@code false} if it was not found.
      *
-     * @throws ConversionException if the resource could not be converted to a {@code boolean}.
+     * @throws javax.application.converter.ConversionException if the resource could not be converted to a {@code boolean}.
      */
     boolean getAsBoolean(String key);
```

Comment:
@aalmiray maybe just simplify the number of methods using generics.

```java
<T> T get(String key, Class<T> clazz);
<T> T get(String key, T defaultValue);
```

Reply:
We've got those methods already. The `getAsX` methods are just a convenience/shortcut for the most common types used by configuration values. To be honest I'd like to reuse JSR-382 but it has not been started yet.